othello/apps/games/consumers.py

```diff
@@ -0,0 +1,108 @@
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import JsonWebsocketConsumer
+
+from django.utils import timezone
+
+from ...moderator.moderator import Player
+from .models import Game, Submission
+from .tasks import run_game
+from .utils import *
+
+
+class GameConsumer(JsonWebsocketConsumer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.game = None
+        self.connected = False
+
+    def connect(self):
+        try:
+            self.game = Game.objects.get(id=self.scope["url_route"]["kwargs"]["game_id"])
+        except Game.DoesNotExist:
+            return self.close()
+
+        try:
+            yourself = Submission.objects.get(user__username="Yourself")
+        except Submission.DoesNotExist:
+            self.close()
+            raise RuntimeError("Cannot find Yourself submission!")
```

Comment:
If you're going to just re-raise an exception, why catch it in the first place?

Reply:
Additionally, if you add `Yourself` in a data migration, this error case should never occur.